File: Weapons/SwordControl.js
// This script control the sword behaviour.
var swordDamage = 1;

function Update (){
    SwordAnimation();
}

function SwordAnimation() {
    var animator = GetComponent(Animator);
    var player = GameObject.Find("Player");
    var getPlayerState = player.GetComponent(Animator).GetCurrentAnimatorStateInfo(0);
    var playerSprite = GameObject.FindWithTag("Player").GetComponent(SpriteRenderer);

    if (playerSprite.flipX) {
        transform.localScale.x = -1;
        transform.localPosition.x = -0.75;
    }
    else {
        transform.localScale.x = 1;
        transform.localPosition.x = 0.75;
    }

    if (Input.GetAxis("Horizontal") && player.GetComponent(PlayerControl).playerCanMove) {
        animator.SetBool("run", true);
    }
    else {
        animator.SetBool("run", false);
    }

    if (player.GetComponent(PlayerControl).isGrounded) {
        animator.SetBool("air", false);
    }
    else {
        animator.SetBool("air", true);
    }

    if (player.GetComponent(PlayerControl).attacking) {
        animator.SetBool("attack", true);
    }
    else {
        animator.SetBool("attack", false);
    }

    if (getPlayerState.IsName("Player_Hurt")) {
        animator.SetBool("hurt", true);
    }
    else {
        animator.SetBool("hurt", false);
    }

    if (getPlayerState.IsName("Player_Die")) {
        animator.SetTrigger("die");
    }

    if (getPlayerState.IsName("Player_Throw")) {
        animator.SetBool("throw", true);
    }
    else {
        animator.SetBool("throw", false);
    }

    if (getPlayerState.IsName("Player_JumpAtk")) {
        animator.SetBool("attack", true);
    }
}

File: Weapons/SwordControl.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class SwordControl : MonoBehaviour {
	Animator animator;
	GameObject player;
	Animator playerAnim;
	string[] animParams;
	
	void Start () {
		animator = GetComponent<Animator>();
		player = GameObject.Find("Player");
		playerAnim = player.GetComponent<Animator>();
	}
	
	void Update () {
		Position();
		Animate();
	}

	void Position() {
		if (player.GetComponent<SpriteRenderer>().flipX) {
			GetComponent<SpriteRenderer>().flipX = true;
			transform.localPosition = new Vector2(-0.74f, transform.localPosition.y);
			GetComponent<BoxCollider2D>().offset = new Vector2(-0.25f, 0.45f);
		} else {
			GetComponent<SpriteRenderer>().flipX = false;
			transform.localPosition = new Vector2(0.74f, transform.localPosition.y);
			GetComponent<BoxCollider2D>().offset = new Vector2(0.25f, 0.45f);
		}
	}

	void Animate() {
		string[] animParams = new string[animator.parameterCount];
		for (int i = 0; i < animator.parameterCount; i++) {
			animParams[i] = animator.GetParameter(i).name;
				if (playerAnim.GetBool(animParams[i])) {
					animator.SetBool(animParams[i], true);
				} else {
					animator.SetBool(animParams[i], false);
				}
		}
	}
}
